components/central-logger/org.wso2.carbon.identity.central.log.mgt/src/main/java/org/wso2/carbon/identity/central/log/mgt/hanlder/CentralLogger.java

```diff
@@ -0,0 +1,63 @@
+/*
+ * Copyright (c) 2021, WSO2 Inc. (http://www.wso2.org).
+ *
+ * WSO2 Inc. licenses this file to you under the Apache License,
+ * Version 2.0 (the "License"); you may not use this file except
+ * in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.wso2.carbon.identity.central.log.mgt.hanlder;
+
+import org.wso2.carbon.identity.core.bean.context.MessageContext;
+import org.wso2.carbon.identity.event.IdentityEventException;
+import org.wso2.carbon.identity.event.event.Event;
+import org.wso2.carbon.identity.event.handler.AbstractEventHandler;
+import org.wso2.carbon.utils.CarbonUtils;
+
+import java.util.Map;
+
+import static org.wso2.carbon.identity.event.IdentityEventConstants.Event.PUBLISH_AUDIT_LOG;
+import static org.wso2.carbon.identity.event.IdentityEventConstants.Event.PUBLISH_DIAGNOSTIC_LOG;
+import static org.wso2.carbon.utils.CarbonUtils.isLegacyAuditLogsDisabled;
+
+/**
+ * Central log event handler for audit, and diagnostic logs.
+ */
+public class CentralLogger extends AbstractEventHandler {
+
+    @Override
+    public boolean isEnabled(MessageContext messageContext) {
+
+        if (super.isEnabled(messageContext)) {
+            return !isLegacyAuditLogsDisabled();
```

Comment:
this validation should be applied only for the audit logs in handleEvent method.

Reply:
addressed in 3687bdc